docs/dev/stream/operators/process_function.md

```diff
@@ -44,8 +44,8 @@ For fault-tolerant state, the `ProcessFunction` gives access to Flink's [keyed s
 The timers allow applications to react to changes in processing time and in [event time]({{ site.baseurl }}/dev/event_time.html).
 Every call to the function `processElement(...)` gets a `Context` object which gives access to the element's
 event time timestamp, and to the *TimerService*. The `TimerService` can be used to register callbacks for future
-event-/processing-time instants. When a timer's particular time is reached, the `onTimer(...)` method is
-called. During that call, all states are again scoped to the key with which the timer was created, allowing
+event-/processing-time instants. The `onTimer(...)` method is
+called when such an event-time is first caught up by a watermark or such a processing-time is reached. During that call, all states are again scoped to the key with which the timer was created, allowing
```

Comment:
I agree we should explain what "when time is reached" means for both event and processing time, but I'm not sure of the actual changes in the PR.

Maybe something along the lines of:
```
When a timer's particular time is reached, the `onTimer(...)` method is called; for processing time, this would mean __________, while for event time, this would mean ______________.
```

Reply:
@alpinegizmo what do you think?

Reply:
I suggest:

With event-time timers, the `onTimer(...)` method is called when the current watermark is advanced up to or beyond the timestamp of the timer, while with processing-time timers, `onTimer(...)` is called when wall clock time reaches the specified time.

Reply:
I don't have a strong opinion about wording as long as it explains things clearly.

Reply:
@an0 can you update the PR to what David suggested?

Reply:
Sure. Done.

Reply:
@tzulitai Is it mergeable?